docs/contributor/arch/004-consolidate-pipeline-statuses.md

```diff
@@ -0,0 +1,56 @@
+Instead, we made our own way by implementing a custom status condition structure, which was a subset of `metav1.Condition`.
+Additionally, instead of treating status collections as a map with a key of type, we treated is as a dynamic list. At the start, a Pending condition was given to the resource upon creation. As underlying resources were deployed and ready, a Running condition was added. If issues arose, we cleared the entire list and only added a Pending condition. This deviates from API best practices and could be confusing for end-users.
+
+That approach was used for `LogPipeline` and `TracePipeline` CRDs. Later on, we realized that we should follow the convention and use `metav1.Condition` instead. We did that for `Telemetry` and `MetricPipeline` CRDs.
+
+### Goal
+
+Standardize the status structure across all Custom Resource Definitions (CRDs) and adhere to API conventions. This will bring consistency in the status structure, making it more straightforward to manage and comprehend for end-users.
+What makes things tricky is that there may be Kyma customers who rely on the old status structure for `LogPipeline` and `TracePipeline` CRDs. We need to ensure that the transition is smooth and that we don't break anything.
+
+## Decision
+
+Roll out the change for `LogPipeline` and `TracePipeline` in multiple phases:
+1. Augment the existing custom condition structures with missing required fields from `metav1.Condition`, like `Status` and `Message`. Make them optional. Extend the controllers' logic to populate them with reasonable values. Such a rollout can be done without bumping `v1alpha1` API group since we will only add optional fields.
+2. Replace custom condition structures with `metav1.Condition`. Technically, it's a breaking change as some optional fields become required. However, we can maintain the API version because the status is automatically reconciled and should never be directly set by a user.
+3. Set new conditions in the status (`GatewayHealthy`/`AgentHealthy`, `ConfigurationGenerated`) and append old conditions to the list (`Pending`, `Running`). In such a way we preserve the semantics since the user would still infer a pipeline healthiness from the last condition in the list. Migrate to new conditions in E2E tests and `Telemetry` state propagation. Deprecate the old conditions and announce it in the Release Notes.
+  ```yaml
+status:
+  conditions:
+  - lastTransitionTime: "2024-02-01T08:26:02Z"
+    message: Trace gateway Deployment is ready
+    observedGeneration: 1
+    reason: DeploymentReady
+    status: "True"
+    type: GatewayHealthy
+  - lastTransitionTime: "2024-01-24T14:36:58Z"
+    message: ""
+    observedGeneration: 1
+    reason: ConfigurationGenerated
+    status: "True"
+    type: ConfigurationGenerated
+  - lastTransitionTime: "2024-02-01T10:43:14Z"
+    message: Trace gateway Deployment is ready
+    observedGeneration: 1
+    reason: DeploymentReady
+    status: "False"
```

Comment:
I think to do not break existing behaviour, we should still add and remove the `Pending` and `Running` condition as before. Otherwise we would break any automation that consumes the last condition in the list. When `Running` is added, we can set the status for the `Pending` condition to False, but I think we must not have a `Running` condition with status: "False" at the end of the list.

Reply:
Yes, it was meant to have the status "True". But you are right, we probably have to add "Pending" and "Running"